src/components/header/header-client.ts

```
@@ -0,0 +1,94 @@
+import { initScrollEffectsManager, type ScrollEffectsManager } from './scroll-effects';
+
+export interface HeaderManagers {
+  theme: ThemeManager;
+  mobileMenu: MobileMenuManager;
+  dropdown: DropdownManager;
+  scrollEffects: ScrollEffectsManager;
+}
+
+export class HeaderClient {
+  private managers: HeaderManagers | null = null;
+  private initialized: boolean = false;
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)